core/src/main/java/org/apache/iceberg/RemoveSnapshots.java

```diff
@@ -161,21 +174,89 @@ public List<Snapshot> apply() {
 
   private TableMetadata internalApply() {
     this.base = ops.refresh();
+    if (base.refs().isEmpty()) {
```

Comment:
What happens when a table without refs is read? Do we create a `main` ref using the `current-snaphot-id`? We need to make sure that this doesn't prevent older metadata from getting cleaned up.

Reply:
Ah yes, good catch, this should actually be base.snapshots().isEmpty() to early return in case there is no snapshots to expire. Currently, we do not create a main branch when we read the metadata. Only when there are snapshots produced do we actually create the main ref at that moment. 

To prevent the situation where a user uses the new procedure on existing metadata where no new snapshots have been produced using the newer library (and thus no main branch has been created), I think just using the snapshot list size as a source of truth should suffice for any early return.